src/Storages/StorageMaterializedView.cpp

```diff
@@ -166,6 +177,17 @@ void StorageMaterializedView::read(
     {
         auto mv_header = getHeaderForProcessingStage(*this, column_names, metadata_snapshot, query_info, local_context, processed_stage);
         auto target_header = query_plan.getCurrentDataStream().header;
+
```

Comment:
Do you have general understanding why `mv_header` and `target_header` may differ and why we add converting actions?
Maybe add some details in comment?

Reply:
There is a comment already - https://github.com/ClickHouse/ClickHouse/pull/24870/files#diff-a598f6628155c97d772f9a5ca630f38e637aec47f9d65bfee48d6e4793471417R186-R188

AFAIR the reason is that Distributed query can be processed up to some stage and this will change output header for queries w/o GROUP BY for instance. Does this elaborate something?